src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java

```diff
@@ -55,6 +55,24 @@
  * <pre>
  * public void foo(final char @NotNull [] param) {} // No violation
  * </pre>
+ * <p>
+ * This check processes the
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_SYNCHRONIZED">
+ * LITERAL_SYNCHRONIZED</a> token only when it appears as a part of a
+ * <a href="https://docs.oracle.com/javase/specs/jls/se19/html/jls-14.html#jls-14.19">
+ * synchronized statement</a>.
+ * </p>
+ * <p>
+ * Example:
```

Comment:
Example above this was already in the description.

Reply:
It looks like the existing example had no issues.

Reply:
It might be good idea to make check on this.
@rnveach , is your xml Checker can handle this ?

Reply:
CS Extras has basic checks for XML. Indentation, Naming, and Illegal Token and Illegal Token Text. If those work, you can use them, otherwise you will need to create an issue for what your asking for.

Reply:
unfortunately we escape all and place it in `<description>` tag
```
 &lt;/pre&gt;
 &lt;p&gt;
 This check processes the
 &lt;a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_SYNCHRONIZED"&gt;
 LITERAL_SYNCHRONIZED&lt;/a&gt; token only when it appears as a part of a
 &lt;a href="https://docs.oracle.com/javase/specs/jls/se19/html/jls-14.html#jls-14.19"&gt;
 synchronized statement&lt;/a&gt;.
 &lt;/p&gt;
 &lt;p&gt;
 Example:
 &lt;/p&gt;
 &lt;pre&gt;
 void m1() {
     synchronized(this) {} // ok
     synchronized (this) {} // violation
 }
 synchronized &amp;lt;T&amp;gt; void m2() {} // ok
 synchronized&amp;lt;T&amp;gt; void m3() {} // ok
 &lt;/pre&gt;</description>
```

xml validation will not work, we might need to update [JavadocMetadataScraper ](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/meta/JavadocMetadataScraper.java)  to not allow `source` tags (there might be already bunch of bad examples that needs to be ignored and fixed separately).
I remember that we were very strict for on this for properties descriptions.
May be for Check descriptions we need more relaxed rules or same. Please share your ideas.

@nrmancuso , while discussion is ongoing, please remove example code and lets merge this fix.

Reply:
>It looks like the existing example had no issues.

@nrmancuso , we need to check how it looks in Sonar or EclipseCS embedded rendering.

Reply:
```
~/java/github/romani/checkstyle [master|✚ 1] 
$ git diff
diff --git a/src/xdocs/config_whitespace.xml b/src/xdocs/config_whitespace.xml
index 55e3ac4..053c32f 100644
--- a/src/xdocs/config_whitespace.xml
+++ b/src/xdocs/config_whitespace.xml
@@ -1347,12 +1347,10 @@ import static java.math.BigInteger.ZERO;
           will be ignored.
         </p>
         <p>
-         If the annotation is between the type and the array, the check will skip validation for
+         If the annotation is between the type and the array like
+         <code>char @NotNull [] param</code>, the check will skip validation for
          spaces
         </p>
-        <source>
-public void foo(final char @NotNull [] param) {} // No violation
-        </source>
       </subsection>
 
       <subsection name="Properties" id="NoWhitespaceAfter_Properties">

```
![image](https://user-images.githubusercontent.com/812984/206466835-27430402-3470-4c16-9213-e0a69a844dc9.png)

Reply:
Issue is opened for check to place violations on source tags in description subsection: https://github.com/rnveach/checkstyle-extras/issues/7

Reply:
Issue is opened to forbid `<pre>` tags in descriptions at https://github.com/checkstyle/checkstyle/issues/12514

Reply:
Done